app/src/main/java/me/ccrama/redditslide/Fragments/WikiPage.java

```diff
@@ -38,66 +61,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             @Override
-            protected Void doInBackground(Void... params) {
-                try {
-                    text = ((Wiki) getActivity()).wiki.get(subreddit, title).getDataNode().get("content_html").asText();
-                } catch(Exception ignored){
-
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                if (url.toLowerCase().startsWith(wikiUrl.toLowerCase()) && listener != null) {
+                    String pagePiece = url.toLowerCase().replace(wikiUrl.toLowerCase(), "")
+                            .split("/")[0]
+                            .split("#")[0];
+                    listener.embeddedWikiLinkClicked(pagePiece);
+                } else {
+                    OpenRedditLink.openUrl(getContext(), url, true);
                 }
-                return null;
+                return true;
             }
 
             @Override
-            protected void onPostExecute(Void aVoid) {
-                setViews(text, subreddit, body, commentOverflow);
-
-                ref.setRefreshing(false);
-                ref.setEnabled(false);
-
+            public void onPageFinished(WebView webView, String url) {
+                super.onPageFinished(webView, url);
+                if (getView() != null) {
+                    getView().findViewById(R.id.wiki_web_view_cover).setVisibility(View.GONE);
+                    ref.setRefreshing(false);
+                    ref.setEnabled(false);
+                }
             }
-        }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
-
-        return v;
+        });
     }
 
-    private void setViews(String rawHTML, String subredditName, SpoilerRobotoTextView firstTextView, CommentOverflow commentOverflow) {
-        if (rawHTML ==null || rawHTML.isEmpty()) {
-            return;
-        }
-
-        List<String> blocks = SubmissionParser.getBlocks(rawHTML);
-
-        int startIndex = 0;
-        // the <div class="md"> case is when the body contains a table or code block first
-        if (!blocks.get(0).equals("<div class=\"md\">")) {
-            firstTextView.setVisibility(View.VISIBLE);
-            firstTextView.setTextHtml(blocks.get(0), subredditName);
-            startIndex = 1;
-        } else {
-            firstTextView.setText("");
-            firstTextView.setVisibility(View.GONE);
-        }
-
-        if (blocks.size() > 1) {
-            if (startIndex == 0) {
-                commentOverflow.setViews(blocks, subredditName);
-            } else {
-                commentOverflow.setViews(blocks.subList(startIndex, blocks.size()), subredditName);
-            }
-        } else {
-            commentOverflow.removeAllViews();
-        }
+    private void onDomRetrieved(String dom) {
+        webView.loadDataWithBaseURL(
+                wikiUrl,
+                "<head>".concat(Wiki.getGlobalCustomCss()).concat("</head>").concat(StringEscapeUtils.unescapeHtml4(dom)),
```

Comment:
For the super huge wikis, the majority of the execution time is spent in unescapeHtml4, moving the call into the WikiAsyncTask doInBackground seems to completely alleviate the main thread blocking